TypeScript error: missing 'player' property on data routes path helper

## Description

`npx tsc --noEmit` on `packages/engine` reports a type error on `main`:

```
packages/engine/src/server/routes/data.ts(48,50): error TS2339: Property 'player' does not exist on type '{ config: string; log: string; legacyLog: string; sceneSummary: (n: number, slug: string) => string; character: (name: string) => string; location: (name: string) => string; locationMap: (name: string, mapId: string) => string; ... 11 more ...; playerNotes: string; }'.
```

The `data.ts` route accesses a `.player` property on the campaign paths helper that doesn't exist in the type definition. Likely either:
- A new path helper needs to be added, or
- The route should use a different existing helper

## Reproduction

```bash
npx tsc --noEmit --project packages/engine/tsconfig.json
```

Pre-existing on `main` as of `85f9a7c`.

--- a/packages/engine/src/server/routes/data.ts
+++ b/packages/engine/src/server/routes/data.ts
@@ -11,7 +11,7 @@ import {
   NotesResponse, NotesUpdateRequest, OkResponse,
   SettingsResponse, SettingsPatch, CostResponse, ErrorResponse,
 } from "@machine-violet/shared";
-import { campaignPaths } from "../../tools/filesystem/scaffold.js";
+import { campaignPaths, machinePaths } from "../../tools/filesystem/scaffold.js";
 
 export const dataRoutes: FastifyPluginAsync = async (server: FastifyInstance) => {
 
@@ -45,7 +45,8 @@ export const dataRoutes: FastifyPluginAsync = async (server: FastifyInstance) =>
 
     // Try characters/<name>.md first, then players/<name>.md
     const paths = campaignPaths(gs.campaignRoot);
-    for (const pathFn of [paths.character, paths.player]) {
+    const mPaths = machinePaths(gs.homeDir);
+    for (const pathFn of [paths.character, mPaths.player]) {
       const path = pathFn(name);
       try {
         if (await fileIO.exists(path)) {

--- a/packages/shared/package.json
+++ b/packages/shared/package.json
@@ -7,7 +7,8 @@
   "types": "dist/index.d.ts",
   "exports": {
     ".": "./dist/index.js",
-    "./*.js": "./dist/*.js"
+    "./*.js": "./dist/*.js",
+    "./types/*.js": "./dist/types/*.js"
   },
   "scripts": {
     "build": "tsc",

